### PR TITLE
Adding a dynamic header that can be used in several pages

### PR DESCRIPTION
**Pull request by maher-suleyman**
PTAL

### CODE REVIEW

File: src/components/common/Header.jsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import React from "react";
+import "./index.css";
+import { Image, Container, Button } from "react-bootstrap";
+import profilePhotoButton from "../../images/profilePhotoButton.png";
+
+const Header = (props) => {
+  const {
+    headerBackground,
+    profilePhoto,
+    pageName /* A string that represents the page name where's the header invoked */,
```
Comment (louisrli) [inline src/components/common/Header.jsx]: put comment above the variable

File: src/components/common/Header.jsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    isOwner,
+    extraComponents,
+    userID,
+  } = props;
+  const [image, setImage] = React.useState(null);
+  const [uploadBtnVisibility, setUploadBtnVisibility] = React.useState(
+    "hidden"
+  );
+  const [hasUploadSucceed, setHasUploadSucceed] = React.useState(false);
```
Comment (louisrli) [inline src/components/common/Header.jsx]: either hasUploadSucceeded or didUploadSucceed

File: src/components/common/Header.jsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+  const handlePhotoInputChange = (e) => {
+    if (e.target.files[0]) {
+      setHasUploadSucceed(true);
+      setImage(e.target.files[0]);
+    }
+  };
+  const handlePhotoUpload = () => {
+    setUploadBtnVisibility("hidden");
+    props.updateProfilePhoto(image, userID);
```
Comment (louisrli) [inline src/components/common/Header.jsx]: userId

**Review by louisrli — APPROVED** [review]

File: src/components/common/Header.jsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            <Image
+              src={profilePhoto}
+              alt="Profile"
+              roundedCircle
+              width="200"
+              height="200"
+            />
+            {/* Check for the owner to allow profile photo changing */}
+            {isOwner && (
+              <div class="photoBtn">
```
Comment (basil-kawak) [inline src/components/common/Header.jsx]: this can be something like ```class="ProfilePhotoBtn"```

**Review by basil-kawak — COMMENTED** [review]
good job LGTM!

**Review by basil-kawak — APPROVED** [review]

**Pull request merged**